src/app/games/martian-chess/MartianChessDummyMinimax.ts

```diff
@@ -0,0 +1,171 @@
+            }
+        }
+    }
+    public getMoveForDrone(x: number, y: number, state: MartianChessState): MartianChessMove[] {
+        const coord: Coord = new Coord(x, y);
+        const diagonalLandingCoords: Coord[] = this.getValidDiagonalCoordForDrone(coord, state);
+        const verticalLandingCoords: Coord[] = this.getValidVerticalCoordForDrone(coord, state);
+        const landingCoords: Coord[] = diagonalLandingCoords.concat(verticalLandingCoords);
+        return this.addMoves(state, coord, landingCoords);
+    }
+    private getValidDiagonalCoordForDrone(startingCoord: Coord, state: MartianChessState) {
+        const diagonalLandingCoords: Coord[] = [];
+        for (const dir of Direction.DIRECTIONS.filter((d: Direction) => d.isDiagonal())) {
+            const landingCoord: Coord = startingCoord.getNext(dir);
+            if (landingCoord.isInRange(4, 8)) {
+                const diagonalIsLegalForDrone: boolean = this.ruler.isDiagonalLegalForDrone(state, dir, startingCoord);
+                if (diagonalIsLegalForDrone) {
+                    diagonalLandingCoords.push(landingCoord);
+                }
+            }
+        }
+        return diagonalLandingCoords;
+    }
+    private getValidVerticalCoordForDrone(coord: Coord, state: MartianChessState) {
+        const verticalLandingCoords: Coord[] = [];
+        for (const dir of Direction.DIRECTIONS.filter((d: Direction) => d.isVertical())) {
+            const landingCoord: Coord = coord.getNext(dir, 2);
+            if (landingCoord.isInRange(4, 8)) {
+                const intermediaryStep: Coord = coord.getNext(dir, 1);
+                if (state.getPieceAt(intermediaryStep) === MartianChessPiece.EMPTY) {
+                    verticalLandingCoords.push(landingCoord);
+                }
+            }
+        }
+        return verticalLandingCoords;
+    }
+    public getMoveForQueen(x: number, y: number, state: MartianChessState): MartianChessMove[] {
+        const startingCoord: Coord = new Coord(x, y);
+        const landingCoords: Coord[] = this.getLandingCoordForQueen(startingCoord, state);
+        return this.addMoves(state, startingCoord, landingCoords);
+    }
+    private getLandingCoordForQueen(startingCoord: Coord, state: MartianChessState): Coord[] {
+        const landingCoords: Coord[] = [];
+        for (const dir of Direction.DIRECTIONS) {
+            let dist: number = 1;
+            let landingCoord: Coord = startingCoord.getNext(dir, dist);
+            let landingContent: MartianChessPiece | null = state.getPieceAtOrNull(landingCoord);
+            let possible: boolean = landingContent === MartianChessPiece.EMPTY;
+            while (possible) {
+                landingCoords.push(landingCoord);
+                dist++;
+                landingCoord = startingCoord.getNext(dir, dist);
+                landingContent = state.getPieceAtOrNull(landingCoord);
+                possible = landingContent === MartianChessPiece.EMPTY;
```

Comment:
hm tout ça serait pas plus clean avec un `for` et deux `let` en moins ?

Reply:
J'utilise encore les let après la boucle du coup je pense pas que ce soit plus clair, puis je suis pas un grand fan du for multiligne j'avoue :D !!